django/core/validators.py

```diff
@@ -471,14 +471,22 @@ class FileExtensionValidator:
     code = 'invalid_extension'
 
     def __init__(self, allowed_extensions=None, message=None, code=None):
-        if allowed_extensions is not None:
-            allowed_extensions = [allowed_extension.lower() for allowed_extension in allowed_extensions]
-        self.allowed_extensions = allowed_extensions
+        self._allowed_extensions = allowed_extensions
         if message is not None:
             self.message = message
         if code is not None:
             self.code = code
 
+    @cached_property
+    def allowed_extensions(self):
```

Comment:
I'm not a fan of the callable or `list` argument, especially for undocumented APIs.

How about leaving the argument out altogether, and doing:
```
if self._allowed_extensions is None:
    allowed_extensions = get_available_image_extensions()
else:
    allowed_extensions = self._allowed_extensions
```

Reply:
The trick is, It's theoretically a generic FileExtensionValidator, not necessarily an ImageFileExtensionValidator.

Reply:
Right! How about a `ImageExtensionValidator` subclass of `FileExtensionValidator` with a `get_default_extension()` method?